client/question/services/FeedbackGeneratorService.js

```diff
@@ -62,6 +121,39 @@ tie.factory('FeedbackGeneratorService', [
     var previousRuntimeErrorString = '';
 
     /**
+     * Pseudo-randomly returns an int between min and max (inclusive).
+     *
+     * @param {number} min Minimum number for range.
+     * @param {number} max Maximum number for range.
+     * @returns {number}
+     * @private
+     */
+    var _getRandomInt = function(min, max) {
+      min = Math.ceil(min);
+      max = Math.floor(max);
+      return Math.floor(Math.random() * (max - min)) + min
+    };
+
+    /**
+     * Randomly selects feedback string from an object containing variations
+     * of a given feedback type.
+     *
+     * @param {Object.<string, Array.<string>>} correctnessFeedbackText
+     * @param {string} type Correctnness feedback type.
+     * @returns {string}
+     * @private
+     */
+    var _selectCorrectnessFeedback = function(correctnessFeedbackText, type) {
```

Comment:
Consider _getCorrectnessFeedbackString; the internal details that this is selection from a random list don't matter to the caller.

Why is correctnessFeedbackText being passed in as a variable, btw? It's global in this scope and I don't think it needs to be passed in explicitly.

Reply:
Changed